src/expression.c

```diff
@@ -9809,6 +9810,69 @@ Expression *SliceExp::syntaxCopy()
     return se;
 }
 
+bool SliceExp::isOpDollar(VarExp* ve)
+{
+    return ve->var == this->lengthVar;
+    // VarDeclaration* decl = ve->var->isVarDeclaration();
+    // return decl && decl->ident == Id::dollar;
+}
+
+SliceExp::Boundness SliceExp::analyzeRelativeBound(Expression* e,
```

Comment:
Why is `lengthVar` a member of `OpSlice`? Shouldn't it be better to have one global instance of it?